include/zephyr/posix/sys/sysconf.h

```diff
@@ -296,6 +331,134 @@ enum {
 #define __z_posix_sysconf_SC_TZNAME_MAX                   TZNAME_MAX
 
 #ifdef CONFIG_POSIX_SYSCONF_IMPL_MACRO
+enum {
```

Comment:
Hmm... having these defined two ways is kind of sketchy.

Reply:
@keith-packard - can you think of any macrobatics that would simplify `CONFIG_POSIX_SYSCONF_IMPL_MACRO`? Some kind of non-evaluative token pasting?

Otherwise, I'm thinking it might be best to go back to the individually numbered enums. Sorry to ask @nslowell to do more work :(

Reply:
I asked myself this several times, but I'm glad you asked again on a new day b/c I discovered that #define sysconf() exists in unistd.h and sysconf.h and I was testing changes on the wrong one.  Preliminary tests show me that we actually want to _avoid_ any double-layer macro, so
`#define sysconf(x) __z_posix_sysconf##x` 
actually seems to work.  I'll make the official change and run through tests again tomorrow.

I appreciate the continued support and patience.

Reply:
Ah, ok - that's how it is now, but we use the `CONCAT()` macro

Reply:
right, we can't use CONCAT() b/c it causes the _SC_* to be pre-expanded.  By concatenting directly in the sysconf() macro, the pre-expansion is avoided.